### PR TITLE
fix/6830

### LINKED ISSUE #6830: pfacct: closed sessions aren't being put in bandwidth_accounting_history
**Describe the bug**
Closed sessions aren't being put in bandwidth_accounting_history

This is at the problematic customer we talked about today (starts with S)

```
MariaDB [pf]> select time_bucket, unique_session_id, last_updated from bandwidth_accounting order by time_bucket limit 10;
+---------------------+----------------------+---------------------+
| time_bucket         | unique_session_id    | last_updated        |
+---------------------+----------------------+---------------------+
| 2021-12-21 00:00:00 | 10917253206297325540 | 0000-00-00 00:00:00 |
| 2021-12-21 00:00:00 |  7342329794988362093 | 0000-00-00 00:00:00 |
| 2021-12-21 00:00:00 |  8112756756300969954 | 0000-00-00 00:00:00 |
| 2021-12-21 00:00:00 |  1030888329191689771 | 0000-00-00 00:00:00 |
| 2021-12-21 00:00:00 |  8894970703260117181 | 0000-00-00 00:00:00 |
| 2021-12-21 00:00:00 | 18132615356821645576 | 0000-00-00 00:00:00 |
| 2021-12-21 00:00:00 |  4539337547764584042 | 0000-00-00 00:00:00 |
| 2021-12-21 00:00:00 |  1778391972636183790 | 0000-00-00 00:00:00 |
| 2021-12-21 00:00:00 | 10522304232321636647 | 0000-00-00 00:00:00 |
| 2021-12-21 00:00:00 |  4187116859012494623 | 0000-00-00 00:00:00 |
+---------------------+----------------------+---------------------+
10 rows in set (0.000 sec)
```

AFAIK this date is the one that the server was upgraded to v11.1 so pfcron is either not rotating the records or not rotating them fast enough into bandwidth_accounting_history

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description
Improve the performance of the bandwidth accounting cleanup process

# Impacts
Bandwidh accounting.

# Issue
fixes #6830

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries

## Enhancements
* Improved the performance of the bandwidth accounting cleanup process

### CODE REVIEW

File: db/upgrade-X.X-X.Y.sql [hunk 1]
```diff
@@ -97,7 +97,9 @@ ALTER TABLE `dhcp_option82_history`
 \! echo "Updating bandwidth_accounting indexes";
 ALTER TABLE bandwidth_accounting
  DROP INDEX IF EXISTS bandwidth_accounting_tenant_id_mac,
- ADD INDEX bandwidth_accounting_tenant_id_mac_last_updated (tenant_id, mac, last_updated);
+ ADD INDEX bandwidth_accounting_tenant_id_mac_last_updated (tenant_id, mac, last_updated),
```
Comment (fdurand) [inline db/upgrade-X.X-X.Y.sql]: ADD INDEX IF NOT EXISTS

Comment (jrouzierinverse) [reply db/upgrade-X.X-X.Y.sql]: This is fixed in devel let me rebase

File: db/upgrade-X.X-X.Y.sql [hunk 2]
```diff
@@ -119,6 +121,211 @@ ALTER TABLE pki_cas
 ALTER TABLE pki_revoked_certs
     ADD COLUMN IF NOT EXISTS `subject` varchar(255) AFTER crl_reason;
 
+DELIMITER /
```
Comment (fdurand) [inline db/upgrade-X.X-X.Y.sql]: Add comment like \! echo "blablabla"

File: db/upgrade-X.X-X.Y.sql [hunk 3]
```diff
@@ -119,6 +121,211 @@ ALTER TABLE pki_cas
...
+                FROM  bandwidth_accounting RIGHT JOIN to_delete_bandwidth_aggregation USING(node_id, time_bucket, unique_session_id);
+        END IF;
+        COMMIT;
+    END tblock;
+
+    SELECT @count AS aggreated;
+END /
+DELIMITER ;
+
+DELIMITER /
```
Comment (fdurand) [inline db/upgrade-X.X-X.Y.sql]: Add comment like \! echo "blablabla"

File: db/upgrade-X.X-X.Y.sql [hunk 4]
```diff
@@ -119,6 +121,211 @@ ALTER TABLE pki_cas
...
+
+        END IF;
+        COMMIT;
+    END tblock;
+
+    SELECT @count AS `count`;
+END/
+DELIMITER ;
+
+DELIMITER /
```
Comment (fdurand) [inline db/upgrade-X.X-X.Y.sql]: Add comment like \! echo "blablabla"

File: db/upgrade-X.X-X.Y.sql [hunk 5]
```diff
@@ -119,6 +121,211 @@ ALTER TABLE pki_cas
...
+
+        END IF;
+        COMMIT;
+    END tblock;
+
+    SELECT @count as `count`;
+END/
+DELIMITER ;
+
+DELIMITER /
```
Comment (fdurand) [inline db/upgrade-X.X-X.Y.sql]: Add comment like \! echo "blablabla"

**Review by fdurand — CHANGES_REQUESTED** [review]

File: conf/pfcron.conf.defaults [hunk 6]
```diff
@@ -743,43 +743,21 @@ history_timeout=10s
...
-#
-# The schedule of task
-schedule=@every 5m
-#
-# batch
+# session_batch
 #
 # How many session to clean up at a time
-batch=100
+session_batch=100
```
Comment (fdurand) [inline conf/pfcron.conf.defaults:750]: needs to be visible from the admin gui

File: conf/pfcron.conf.defaults [hunk 7]
```diff
@@ -743,43 +743,21 @@ history_timeout=10s
...
 #
-# timeout
+# session_timeout
 #
 # How long a session cleanup job can run
-timeout=5s
+session_timeout=5s
```
Comment (fdurand) [inline conf/pfcron.conf.defaults:755]: needs to be visible from the admin gui

File: conf/pfcron.conf.defaults [hunk 8]
```diff
@@ -743,43 +743,21 @@ history_timeout=10s
...
 #
-# window
+# session_window
 #
 # How long to keep a session open
-window=24h
+session_window=24h
```
Comment (fdurand) [inline conf/pfcron.conf.defaults:760]: needs to be visible from the admin gui

**Review by fdurand — CHANGES_REQUESTED** [review]
Remove the procedure and fix the missing admin gui parameters and we will be good to merge.

**Pull request merged**